subcommand/mesh-init/command.go

```diff
@@ -467,6 +495,32 @@ func (c *Command) generateAndWriteDataplaneConfig(proxyRegistration *api.Catalog
 	return nil
 }
 
+func (c *Command) deregisterServiceAndProxy(consulClient *api.Client, clusterARN string, serviceRegistration, proxyRegistration *api.CatalogRegistration) error {
+	if serviceRegistration != nil {
+		deregisterConsulService(consulClient, serviceRegistration, clusterARN)
+	}
+
+	// Proxy deregistration
+	deregisterConsulService(consulClient, proxyRegistration, clusterARN)
+	return nil
```

Comment:
it looks like we never return the error here. Should this be the following?

```suggestion
	if serviceRegistration != nil {
		return deregisterConsulService(consulClient, serviceRegistration, clusterARN)
	}

	// Proxy deregistration
	return deregisterConsulService(consulClient, proxyRegistration, clusterARN)
```